pkg/controller/component/vars.go

```diff
@@ -340,6 +340,9 @@ func resolveClusterObjectRefVars(ctx context.Context, cli client.Reader, synthes
 	}
 	vars1, vars2 := make([]corev1.EnvVar, 0), make([]corev1.EnvVar, 0)
 	for _, v := range definedVars {
+		if len(v.Value) > 0 && v.ValueFrom != nil {
```

Comment:
Perform this check in the cmpd controller.

Reply:
> Perform this check in the cmpd controller.

done